go/private/actions/archive.bzl

```diff
@@ -129,12 +138,28 @@ def emit_archive(go, source = None, _recompile_suffix = "", recompile_internal_d
             archives = direct,
             out_lib = out_lib,
             out_export = out_export,
+            out_nogo_log = out_nogo_log,
             gc_goopts = source.gc_goopts,
             cgo = False,
             testfilter = testfilter,
             recompile_internal_deps = recompile_internal_deps,
         )
 
+    if go.nogo:
+        validations.append(out_nogo_validation)
+        go.actions.run_shell(
+            inputs = [out_nogo_log],
+            outputs = [out_nogo_validation],
+            # Create the expected output file if there are no errors, otherwise print the errors to stderr.
+            command = '[ -s "$1" ] && cat <(echo) "$1" <(echo) 1>&2 || touch $2',
```

Comment:
I don't like this approach. The actual nogo is still running as part of GoCompilePkg action, producing an output that is consumed by this no-ops command.

I would prefer if you start with separating `nogo` part of the go code into a separate `builder` sub-command, or ever a separate binary. Then in the validation action, we instrument that new subcommand/binary instead.

Not quite sure if it's a thing, but ideally user should be able to request for a `--output_group=_validation` and only trigger the validation actions to run linter, without much dependency on the compilation action.